### ISSUE TITLE
it appears Serrated when use SAO

### ISSUE THREAD

**Issue by trymoremore** [mrdoob/three.js#12085]
i use the SAO and set the render as this :

```javascript
renderer = new THREE.WebGLRenderer({
   antialias: true,       
   precision: "highp",     
   alpha:false       
} );

renderer.setSize( window.innerWidth, window.innerHeight );
renderer.setClearColor(0xffffff);
renderer.setPixelRatio( window.devicePixelRatio );
document.body.appendChild( renderer.domElement );

composer = new THREE.EffectComposer(renderer);
renderPass = new THREE.RenderPass(scene, camera);
composer.addPass(renderPass);

saoPass = new THREE.SAOPass(scene, camera, false, true);
saoPass.renderToScreen = true;
composer.addPass(saoPass);
```
however it appears serrated. 
whether the serrated would be appear when using sao,or my setting is something wrong?

**Comment by Mugen87** [comment]
Can you share a screenshot or a live demo?

You might want to use a `FXAA` at the end of your pass chain, see

https://github.com/mrdoob/three.js/blob/master/examples/webgl_postprocessing_outline.html#L306-L309

**Comment by trymoremore** [comment]
![image1](https://user-images.githubusercontent.com/27277007/29867727-11fc49c8-8daf-11e7-85b9-60fda09bd96c.png) it is use SAO and choose Beauty.
![image](https://user-images.githubusercontent.com/27277007/29867829-5eea0c0c-8daf-11e7-85a9-3ed89809bebd.png)it is normal and without SAO

**Comment by Mugen87** [comment]
Have you tried an additional FXAA pass for anti-aliasing?

**Comment by Mugen87** [comment]
https://threejs.org/examples/?q=postprocessing

**Comment by Usnul** [comment]
TL;DR: working as intended, limitation of WebGL

this is working as intended. "serrated" effect you are referring to are aliasing artifacts, when you use post-processing - you render not to screen but to a VFB (virtual frame buffer) and WebGL does not support anti-aliasing for this rendering mode, as a result you end up having to use a separate anti-aliasing pass, such as FXAA that @Mugen87 mentioned. There are a few other anti-aliasing shaders available in examples folder that you might want to look at.

**Comment by trymoremore** [comment]
@Mugen87 @Usnul thanks for your help！i  use a FXAA pass at the end of my pass chain,but it seems dosen't work ,and the effect of SAO is disappeared.this is my 'init()' code:
`function init(){
    //scene
    scene = new THREE.Scene();

    //camera
    camera = new THREE.PerspectiveCamera( CAMERA.fov, window.innerWidth / window.innerHeight, CAMERA.near, CAMERA.far);
    camera.position.set(CAMERA.cameraPosition.X, CAMERA.cameraPosition.Y, CAMERA.cameraPosition.Z);
    camera.scale.set(CAMERA.cameraScale.X,CAMERA.cameraScale.Y,CAMERA.cameraScale.Z);

    //light
    for(var lig in lights)
    {
        var lightNow;
        switch(lights[lig].lightType)
        {
            case "AmbientLight":
                lightNow = new THREE.AmbientLight( lights[lig].lightColor );
                break;
            case "DirectionalLight":
                lightNow = new THREE.DirectionalLight( lights[lig].lightColor );
                break;
            case "HemisphereLight":
                lightNow = new THREE.HemisphereLight( lights[lig].lightColor );
                break;
            case "PointLight":
                lightNow = new THREE.PointLight( lights[lig].lightColor );
                break;
            case "SpotLight":
                lightNow = new THREE.SpotLight( lights[lig].lightColor );
                break;
            default:
                break;
        }
        lightNow.position.set(lights[lig].lightPosition.X,lights[lig].lightPosition.Y,lights[lig].lightPosition.Z);
        lightNow.intensity = lights[lig].lightIntensity;
        lightArray[lig] = lightNow;
        scene.add(lightNow);
    };

    //renderer
    renderer = new THREE.WebGLRenderer({
        antialias:false,       
        precision:"highp",  
        alpha:false,          
    } );
    renderer.setSize( window.innerWidth, window.innerHeight );
    renderer.setClearColor(0xffffff);
    renderer.setPixelRatio( window.devicePixelRatio );
    document.body.appendChild( renderer.domElement );
    var controls = new THREE.OrbitControls( camera, renderer.domElement );
    controls.maxPolarAngle = Math.PI * control.controlsMaxPolarAnglePI;
    controls.minDistance = control.controlsMinDistance;
    controls.maxDistance = control.controlsMaxDistance;
    for (var i = lightArray.length - 1; i >= 0; i--) {
        var control2=new THREE.OrbitControls(lightArray[i],renderer.domElement);
        control2.maxPolarAngle = Math.PI * control.controlsMaxPolarAnglePI;//设置转角范围
    };
    //SAO
     composer = new THREE.EffectComposer(renderer);
     renderPass = new THREE.RenderPass(scene, camera);
     composer.addPass(renderPass);
       saoPass = new THREE.SAOPass(scene, camera, false, true);
     saoPass.renderToScreen = true;
     composer.addPass(saoPass);
     outlinePass = new THREE.OutlinePass( new THREE.Vector2(window.innerWidth, window.innerHeight), scene, camera);
      composer.addPass( outlinePass );
    effectFXAA = new THREE.ShaderPass(THREE.FXAAShader);
    effectFXAA.uniforms['resolution'].value.set(1 / window.innerWidth, 1 / window.innerHeight );
    effectFXAA.renderToScreen = true;
   composer.addPass( effectFXAA );
    var gui=fxaaui();
     //var gui=saoui();
};`
the screenshot：
![imagefxaa](https://user-images.githubusercontent.com/27277007/29904897-ca964dba-8e3d-11e7-8c2f-889991ef624b.png)

**Comment by trymoremore** [comment]
@Mugen87 @Usnul the effect of shadow is like this：
before：
![image](https://user-images.githubusercontent.com/27277007/29904967-493b5f20-8e3e-11e7-9189-4270e71345d1.png)
after：
![imageafter](https://user-images.githubusercontent.com/27277007/29905030-aaffa70c-8e3e-11e7-8d62-38a9a4c7ed5a.png)
i know SAO and SSAO can do it,could you tell me other way can achieve this effect

**Comment by trymoremore** [comment]
@Mugen87 @Usnul thank you very much！my problem is slove
